### DISCUSSION TITLE
Allow rule, but notify?

### DISCUSSION THREAD

**Discussion by jonas-w** [evilsocket/opensnitch#1110] [Q&A]
Is it possible to create an allow rule, which allows the connections automatically, but still notifies me?

There are some domains which I don't want to press allow all the time, but I still want to know if they are accessed.

For example github.com, 99% of the time I don't care that something accesses github.com, but I don't want to allow it completely as I will not know when and if something accesses github.com.

**Comment by gustavo-iniguez-goya** [comment]
This is not possible at the moment.

It could be implemented in the daemon as a new field of the rules, "Actions" for example, where the action to be executed could be "notify", "exec", etc.

```
Actions: [
 {
   type: notify
   data: "warning: {DST_HOST} connection attempt"
 },
 {
   type: exec:
   data: /path/script.sh {DST_IP} {DST_PORT} {PROC_PATH} ...
 }
}
```

"notify" would fire a new Alert from the daemon to the GUI.
"exec" would just execute a command.


Or on the server (GUI). I worked on this option some time ago and I hope to publish a branch with the changes. It remains to develop the dialogs to configure the actions easily.

Both options have their pros and cons.
